source/grid/manifold.cc

```diff
@@ -44,19 +44,34 @@ project_to_manifold (const std::vector<Point<spacedim> > &,
   return Point<spacedim>();
 }
 
-
+template <int dim, int spacedim>
+Point<spacedim>
+Manifold<dim, spacedim>::
+get_new_point (const Point<spacedim> &P1, const Point<spacedim> &P2, double w) const
+{
+  return w*P1 + (1-w)*P2;
+}
 
 template <int dim, int spacedim>
 Point<spacedim>
 Manifold<dim, spacedim>::
 get_new_point (const Quadrature<spacedim> &quad) const
 {
+  Assert(quad.size() > 0,
+         ExcMessage("Quadrature should have at least one point."));
+
   Assert(std::abs(std::accumulate(quad.get_weights().begin(), quad.get_weights().end(), 0.0)-1.0) < 1e-10,
          ExcMessage("The weights for the individual points should sum to 1!"));
 
-  Point<spacedim> p;
-  for (unsigned int i=0; i<quad.size(); ++i)
-    p += quad.weight(i) * quad.point(i);
+  Point<spacedim> p = quad.point(0);
+  double w = quad.weight(0);
+
+  for (unsigned int i=1; i<quad.size(); ++i)
+    {
+      if ( w != 0 && quad.weight(i) != 0)
+        p = get_new_point(p, quad.point(i) , w/(quad.weight(i) + w) );
+      w += quad.weight(i);
+    }
 
   return project_to_manifold(quad.get_points(), p);
```

Comment:
This instead should be just `p`. get_new_point` always returns points on the Manifold